### ISSUE TITLE
Race condition, service cannot be started as it is not available

### ISSUE THREAD

**Issue by vanessalove** [OctopusDeploy/Issues#1649]
```
Get-WMIObject : Provider load failure 
At C:\Program Files\Octopus Deploy\Tentacle\Scripts\Octopus.Features.WindowsService_BeforePostDeploy.ps1:125 char:11
+ $status = Get-WMIObject win32_service -filter ("name='" + $serviceName + "'") -c ...
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : InvalidOperation: (:) [Get-WmiObject], ManagementException
    + FullyQualifiedErrorId : GetWMIManagementException,Microsoft.PowerShell.Commands.GetWmiObjectCommand
```

Appears to occur when a service is created, and then a start is attempted but it isn't available yet. We had the customer add a sleep between the create and start commands and it worked each time instead of 'sometimes'.

Can we add a try again when this error occurs.

Source: http://help.octopusdeploy.com/discussions/problems/34742

**Comment by PaulStovell** [comment]
Not 3.0 specific so might wait for a patch

**Comment by PaulStovell** [comment]
This should be fixed by #1911
